submissions/morenko/task-frogger/js/app.js

```diff
@@ -0,0 +1,177 @@
+const countRows = 6;
+const countColumns = 5;
+
+const tileHight = 83;
+const tileWidth = 105;
+
+class Position {
+    startX;
+    startY;
+
+    x;
+    y;
+
+    minY;
+    minX;
+    maxY;
+    maxX;
+    
+    constructor( row, col ) {
+        
+        this.startX = (tileWidth * col) - (tileWidth);
+        this.startY = (tileHight * row) - (tileHight / 2);
+        this.x = this.startX;
+        this.y = this.startY;
+
+        this.maxY = tileHight * (countRows - 1);
+        this.maxX = tileWidth * (countColumns - 1);
+        this.minY = tileHight * (row - 1) - this.startY;
+        this.minX = 0;
+    }
+
+    reset() {
+        this.x = this.startX;
+        this.y = this.startY;
+        
+    }
+}
+
+class Enemy {
+
+    speed = (Math.floor(Math.random(4) * (500 - 100 + 1)) + 100);
+    sprite = 'images/Rock.png';
+
+    constructor ( PositionEnemy ) {
+        this.position = PositionEnemy;
+    }
+
+    update (dt) {
+        if (this.position.x <= this.position.maxX) {
+            this.position.run(this.speed, dt);
+        } else {
+            this.position.reset();
+        }
+    }
+
+    render = function() {
+        ctx.drawImage(Resources.get(this.sprite), this.position.x, this.position.y);
+    }
+}
+
+class PositionEnemy extends Position {
+
+    run(speed, dt) {
+        this.x += (speed * dt); 
+    }
+}
+
+class Player {
+    sprite = 'images/char-princess-girl.png';
+
+    constructor ( PositionPlayer ) {
+        this.position = PositionPlayer;
+    }
+
+    alertWin() {
+        if (this.position.y == this.position.minY) {
+            setTimeout(() => {
+                alert("Not bed! Try again!");
+                this.position.reset();
+            }, 0);
+        }
+    }
+
+    update () {
+        this.alertWin();
```

Comment:
This doesn't seem like a good usage of naming, isn't it? This function's name says `alertWin()`. Does it do that? Does it alert about a win when called? No. It checks if the win condition is met and only then alerts about win. So you have to either take the check out of this function and leave only alerting to it, or change the name to something like `checkForWin`, because that's what this function is doing.

Reply:
Yes, you are right